File: frontend/src/models/project.ts
export interface ProjectData {
  id: string
  ownerId: string
  title: string
  slug: string
  description: string
  repo: string
  workItems: { total: number; active: number; resolved: number }
  agents: { total: number; running: number }
  lastActivity: string
  branchPattern: string
  commitAuthorMode: string
  commitAuthorName?: string | null
  commitAuthorEmail?: string | null
}

export interface ProjectBranch {
  name: string
  isDefault: boolean
  isProtected: boolean
  canUseForDynamicIteration: boolean
  dynamicIterationBlockedReason?: string | null
}

export interface SlugCheckResult {
  slug: string
  available: boolean
}

File: Fleet.Server/Data/Entities/Project.cs
namespace Fleet.Server.Data.Entities;

public class Project
{
    public string Id { get; set; } = string.Empty;
    public string OwnerId { get; set; } = "default";
    public string Title { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Repo { get; set; } = string.Empty;
    public string LastActivity { get; set; } = string.Empty;
    public string BranchPattern { get; set; } = "fleet/{workItemNumber}-{slug}";
    public string CommitAuthorMode { get; set; } = "fleet";
    public string? CommitAuthorName { get; set; }
    public string? CommitAuthorEmail { get; set; }

    // Stored as JSON (jsonb) columns — demonstrates PostgreSQL JSON support
    public WorkItemSummary WorkItemSummary { get; set; } = new();
    public AgentSummary AgentSummary { get; set; } = new();

    // Navigation properties
    public List<WorkItem> WorkItems { get; set; } = [];
    public List<WorkItemLevel> WorkItemLevels { get; set; } = [];
    public List<ChatSession> ChatSessions { get; set; } = [];
    public List<AgentExecution> AgentExecutions { get; set; } = [];
    public List<LogEntry> LogEntries { get; set; } = [];
    public List<DashboardAgent> DashboardAgents { get; set; } = [];
    public List<MemoryEntry> MemoryEntries { get; set; } = [];
}
